contracts/truefi3/interface/IDebtInstrumentWithSchedule.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {IDebtInstrument} from "./IDebtInstrument.sol";
+
+/**
+ * @title Debt Instrument With Schedule
+ * @notice TrueFi Debt Instrument with Payment Schedule
+ * @dev This Debt Instrument stores an expected schedule of payment
+ * Calling the repay() function should store time and amount of payments
+ * Minting the token should pass in a schedule which is an array
+ * of information about expected payment times and amounts
+ **/
+interface IDebtInstrumentWithSchedule is IDebtInstrument {
+    // store payment and schedule data
+    struct Payment {
+        // time of payment
+        uint256 timestamp;
+        // payment amount
+        uint256 amount;
+    }
+
+    /// @dev Get expected schedule of payments
+    function schedule() external view returns (Payment[] memory);
```

Comment:
A few things here:
1) `tokenId` arg seems to be missing
2) we should not return an array, rather have sth like `nextPayment()`, as:
- you can't iterate it on the blockchain
- whatever calculations it does can be done on front-end

Reply:
We want to be able to get the full schedule to know if a debt is defaulted or not.

This is inherently not scalable past a certain number of payments and schedules. A better way might be to pass in a place in the array we want to get data for instead of returning the whole array. Thoughts?

Reply:
Passing in an array access index could be a viable compromise, although that still forces us to use arrays in the backing storage. Seemingly trivial operations like accessing the total amount currently due would be stuck behind linear iteration, and finding the first date that satisfies X property would be a logarithmic binary search. Modifying schedules (if we ever decide we need to) would be a linear expense as well.

Our largest concern here is unbounded array access. OpenZeppelin has [written a doc](https://defender.openzeppelin.com/#/advisor/docs/access-arrays-using-enumeration-and-pagination) that rates this as a Critical severity issue with a Medium effort solution. Can we can get the desired behavior via a less dangerous avenue?
- @vanruch prefers a `nextPayment()` access function for _just_ the next payment.
- @barrutko mentioned extracting this to a function that calculate deadlines based on a timestamp (so my guess is it would look like `nextPayment(uint256 timestamp)`?)

If we need to display an expensive view on the frontend, then @vanruch's proposal is to expose variables for that view, but don't add it to the core on-chain interface. @barrutko's proposal should natively be able to give all the info that a frontend would need.